### DISCUSSION TITLE
Rustdesk is unable to compile/build on windows 10 & Ubuntu. Magnum-Opus & Scrap library error.

### DISCUSSION THREAD

**Discussion by 997123** [rustdesk/rustdesk#2860] [Q&A]
![photo_2023-01-18_12-20-58](https://user-images.githubusercontent.com/122609560/213158719-cf8a88ec-1230-4201-a018-24ecadd70928.jpg)
![photo_2023-01-18_12-20-52](https://user-images.githubusercontent.com/122609560/213158725-6b7d235a-8db7-474e-a0eb-0322ee6aebda.jpg)

**Photos** seen above represent two similar problems with which i have  fought for few days already. I have tried two approaches, first on windows 10 64 bit, second on Ubuntu 20.04 64bit.  The issues are very alike due to the fact that in both situations the origin of error is probably the same. I suspect that the main issue is scrap itself.

I need help with compiling and building rustdesk for windows and linux.  I have tried to handle it myself, with little but effect.

Here is what i have done in exact order in which it was done in both cases.

On Windows: (https://rustdesk.com/docs/en/dev/build/windows/)
1.
git clone https://github.com/microsoft/vcpkg
  cd vcpkg
  git checkout 2021.12.01
  cd ..
  vcpkg/bootstrap-vcpkg.bat
  export VCPKG_ROOT=$PWD/vcpkg
  vcpkg/vcpkg install libvpx:x64-windows-static libyuv:x64-windows-static opus:x64-windows-static
2.  add System environment variable VCPKG_ROOT=<path>\vcpkg
3. I have downloaded scritter.dll
4. add System environment variable LIBCLANG_PATH=<llvm_install_dir>/bin
5. git clone https://github.com/rustdesk/rustdesk
cd rustdesk
mkdir -p target/debug
mv sciter.dll target/debug
6. cargo run

On step 6th, i have encoutered an error which was presented above. I tried to solve it. The only informations i could find was to check whether libraries were properly installed. In fact, they were properly installed. I even reinstalled everything and did anything necessary to make it work. But unfortunately, it didn't.

Then i decided to try it on linux. I was using oracle virtual box to install ubuntu. Yet again,  here i present what i have done in exact order in which it was done.

Linux Ubuntu 20.04 64bit (https://rustdesk.com/docs/en/dev/build/linux/)

1.sudo apt install -y g++ gcc git curl wget nasm yasm libgtk-3-dev clang libxcb-randr0-dev libxdo-dev
libxfixes-dev libxcb-shape0-dev libxcb-xfixes0-dev libasound2-dev libpulse-dev cmake
2.git clone https://github.com/microsoft/vcpkg
cd vcpkg
git checkout 2021.12.01
cd ..
vcpkg/bootstrap-vcpkg.sh
export VCPKG_ROOT=/vcpkg
vcpkg/vcpkg install libvpx libyuv opus
3. Here another wild error appeared:
![photo_2023-01-18_12-24-50](https://user-images.githubusercontent.com/122609560/213163157-314cd142-79bb-4d88-86ce-12f3ffd6eb16.jpg)
I have managed to solve it by using : sudo apt-get install libgstreamer1.0-dev libgstreamer-plugins-base1.0-dev
4.curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
source $HOME/.cargo/env
git clone https://github.com/rustdesk/rustdesk
cd rustdesk
mkdir -p target/debug
wget https://raw.githubusercontent.com/c-smile/sciter-sdk/master/bin.lnx/x64/libsciter-gtk.so
mv libsciter-gtk.so target/debug
VCPKG_ROOT=$HOME/vcpkg cargo run
5.  After using cargo run it returns me error presented above.

I would appreciate any form of help, it is really important to me to solve this problem.

**Comment by OMegaComp** [comment]
Did you ever figure this out? on Windows?
I am getting the same errors(Building RD) but trying to install these "libvpx:x64-windows-static libyuv:x64-windows-static"(I think both would) Hang at "Generating MSBuild".
I did not realize at first and tried to to build RD with cargo and got the same file not found error of vp8.h missing.
I cannot seem to get the libraries to compile except for opus.

I have vcpkg environment variable set, even reinstalled about everything.

Any way to get these libraries on? I want to deploy my version of RD on PC's I service for people that need assistence and I got one going out soon I wanted to put it on.
Thanks whoever has info.  :0)

**Comment by rustdesk** [comment]
https://github.com/rustdesk/rustdesk/discussions/763, go to [discord](https://discord.gg/nDceKgxnkV) please, search there first, some users might have asked similar questions. Usually nobody will help you with build or self-hosting here.
